lib/ddtrace/sampler.rb

```diff
@@ -93,37 +93,49 @@ class PrioritySampler
     extend Forwardable
 
     def initialize(opts = {})
-      @base_sampler = opts[:base_sampler] || RateSampler.new
       @post_sampler = opts[:post_sampler] || RateByServiceSampler.new
     end
 
     def sample(span)
-      return perform_sampling(span) unless span.context
-      return sampled_by_upstream(span) if span.context.sampling_priority
-
-      perform_sampling(span).tap do |sampled|
-        span.context.sampling_priority = if sampled
-                                           Datadog::Ext::Priority::AUTO_KEEP
-                                         else
-                                           Datadog::Ext::Priority::AUTO_REJECT
-                                         end
+      # If we haven't sampled this trace yet, do so.
+      # Otherwise we want to keep whatever priority has already been assigned.
+      unless sampled_by_upstream(span)
+        # Use the underlying sampler to "roll the dice" and see how we assign it priority.
+        # This sampler derives rates from the agent, which updates the sampler's rates
+        # whenever traces are submitted to the agent.
+        perform_sampling(span).tap do |sampled|
```

Comment:
Since we're no longer doing any sampling here - Would renaming this to `assign_priority(span)` or something similar make sense here ?

Reply:
Renamed.